lib/navigation/preprocess_encoders.cpp

```diff
@@ -34,8 +38,34 @@ core::EncoderData EncodersPreprocessor::detectOutliers(const core::EncoderData e
   -also figure out return type/ what we update and update
   documentation as appropriate
   */
-  return {0, 0, 0, 0};
-}
+ //core::EncoderData encoder_data_copy = std::copy(encoder_data.begin(), encoder_data.end(), std::back_inserter(encoder_data));
+ //core::EncoderData encoder_data_copy = std::copy(encoder_data.begin(), encoder_data.end())
+ core::EncoderData encoder_data_copy;
+ core::Float median = 0;
+ for(int i = 0;i<core::kNumEncoders;i++){
+  encoder_data_copy.at(i) = encoder_data.at(i);
+ }
+ if(encoder_data.size() % 2 == 0){
+    median =  ((encoder_data.at(encoder_data.size()/2) + encoder_data.at((encoder_data.size()/2) + 1))/2.0);  //calculating the median of encoder data if the length of the array is even
+                                                                                                               
+  }
+  else{
```

Comment:
repeated functionality. Define a helper function.

Reply:
Hint: Look at @BenET01's pr. There is a helper function to get the quartiles and the use of the `Quartiles` struct as a return type.